src/main/java/com/mineinabyss/mobzy/ecs/goals/mobzy/hostile/ThrowItemsGoal.kt

```diff
@@ -3,45 +3,43 @@ package com.mineinabyss.mobzy.ecs.goals.mobzy.hostile
 import com.mineinabyss.idofront.destructure.component1
 import com.mineinabyss.idofront.destructure.component2
 import com.mineinabyss.idofront.destructure.component3
-import com.mineinabyss.idofront.serialization.SerializableItemStack
 import com.mineinabyss.mobzy.api.helpers.entity.distanceSqrTo
 import com.mineinabyss.mobzy.api.nms.aliases.toNMS
 import com.mineinabyss.mobzy.api.pathfindergoals.doneNavigating
 import com.mineinabyss.mobzy.api.pathfindergoals.moveToEntity
 import com.mineinabyss.mobzy.api.pathfindergoals.stopNavigation
 import com.mineinabyss.mobzy.ecs.components.initialization.pathfinding.PathfinderComponent
+import com.mineinabyss.mobzy.ecs.serializers.MobTypeSerializer
+import com.mineinabyss.mobzy.mobs.MobType
+import com.mineinabyss.mobzy.mobs.types.ProjectileEntity
 import com.mineinabyss.mobzy.pathfinders.MobzyPathfinderGoal
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
-import kotlinx.serialization.Transient
-import net.minecraft.server.v1_16_R2.*
 import org.bukkit.Sound
-import org.bukkit.craftbukkit.v1_16_R2.inventory.CraftItemStack
 import org.bukkit.entity.Creature
 import org.bukkit.entity.LivingEntity
 import org.bukkit.entity.Mob
-import org.bukkit.inventory.ItemStack
 import kotlin.math.min
 import kotlin.math.pow
 import kotlin.random.Random
 
 @Serializable
 @SerialName("mobzy:behavior.throw_items")
 class ThrowItemsBehavior(
-    @SerialName("item") private val _item: SerializableItemStack,
-    val damage: Float,
+    val type: @Serializable(with = MobTypeSerializer::class) MobType,
```

Comment:
I implemented the mob type serializer, would like to improve how we do this later but this should be ok for now